src/main/java/org/wildfly/security/audit/RotatingFileAuditEndpoint.java

```diff
@@ -50,7 +50,7 @@
 
     private String nextSuffix;
     private long nextRollover = Long.MAX_VALUE;
-    private volatile long currentSize = 0;
```

Comment:
This is not a great solution I think, as it relies on synchronization by external parties.  Unless the class as a whole does not have a thread-safety contract established, it's better to replace this with an AtomicLong which has an atomic getAndAdd method on it.  Same for the other method below.

On the other hand if the class as a whole needs to update multiple fields atomically, then locking is necessary and should be done directly within the class IMO.